schemas/manifest.schema.json

```
@@ -0,0 +1,455 @@
+                            "type": "string",
+                            "pattern": "^[^:]*(:[rwx-]{3}$)?"
+                        }
+                    }
+                },
+                "permissions": {
+                    "type": "object",
+                    "required": [],
+                    "additionalProperties": false,
+                    "patternProperties": {
+                        "^.[a-z][a-z0-9_]*$": {
+                            "type": "object",
+                            "required": ["url"],
+                            "additionalProperties": false,
+                            "properties": {
+                                "url": {
+                                    "type": "string",
+                                    "pattern": "^/.*$"
+                                },
+                                "show_title": {
+                                    "type": "boolean"
+                                },
+                                "allowed": {
+                                    "type": "string"
+                                },
+                                "auth_header": {
+                                    "type": "boolean"
+                                },
+                                "protected": {
+                                    "type": "boolean"
+                                },
+                                "additional_urls": {
+                                    "type": "array",
+                                    "items": {
+                                        "type": "string"
+                                    }
+                                }
+                            }
+                        }
+                    }
+                },
+                "ports": {
+                    "type": "object",
+                    "required": [],
+                    "additionalProperties": false,
+                    "patternProperties": {
+                        "^.[a-z][a-z0-9_]*$": {
+                            "type": "object",
+                            "required": [],
+                            "additionalProperties": false,
+                            "properties": {
+                                "default": {
+                                    "type": "integer"
+                                },
+                                "exposed": {
+                                    "type": "boolean"
+                                },
+                                "fixed": {
+                                    "type": "boolean"
+                                }
+                            }
+                        }
+                    }
+                },
+                "sources": {
+                    "type": "object",
+                    "required": [],
+                    "additionalProperties": false,
+                    "patternProperties": {
+                        "^[a-z][a-z0-9_]*$": {
```

```suggestion
                        "^[a-z][a-z0-9_-]*$": {
```

Some sources have `-` in their names

---

*gasp* Alright, I have put a lot of `"^[a-z][a-z0-9_]*$"` in the file. Should i change all of them ? Or only this one ?

---

Hmmmm idk, not hard opinion on this, I'm mainly trying to check what's causing issues in existing TOML :grimacing: